### PR TITLE
Configure,util/shlib_wrap.sh: harmonize -Wl and -rpath handling.

### PR DESCRIPTION
**Pull request by dot-asm**
Addresses parts of #6500.

### CODE REVIEW

File: util/shlib_wrap.sh.in [hunk 1]
```diff
@@ -101,7 +101,7 @@ SunOS|IRIX*)
 	;;
 esac
 
-{- output_off() if $config{ex_libs} !~ /,-rpath,/; ""; -}
+{- output_off() unless grep (/-rpath/, @{$config{LDFLAGS}}); ""; -}
```
Comment (dot-asm) [inline util/shlib_wrap.sh.in]: Just in case for reference, see related https://github.com/openssl/openssl/issues/6500#issuecomment-398714504

**Comment by dot-asm** [comment]
Additional commit that adds rpath_flag is not nominated for approval, it's presented to give a taste. Well, it doesn't mean that it can't be approved [if suggestion to have Configure recognize e.g. --run-path will be considered worthy.]

**Comment by levitte** [comment]
I cannot see myself approving this unless you move the `rpath_flag` commit to a different PR.

**Comment by dot-asm** [comment]
I did say that I'm not nominating it for approval, i.e. one could as well say "approve without add-on". But I squashed and removed add-on...

**Comment by levitte** [comment]
I know you said that, just want to avoid possible last minute mistakes.

**Pull request closed without merging**